google/cloud/bigquery/job/query.py

```diff
@@ -1507,6 +1514,15 @@ def result(  # type: ignore  # (incompatible with supertype)
                 # Intentionally omit job_id and query_id since this doesn't
                 # actually correspond to a finished query job.
             )
+
+        # When timeout has default sentinel value, only pass the sentinel
+        # timeout to QueryJob.done(), and use None for the other timeouts.
+        if type(timeout) is object:
+            timeout = None
+            get_job_timeout = POLLING_DEFAULT_VALUE
```

Comment:
Why not set to `DEFAULT_GET_JOB_TIMEOUT` or do the kwargs trick to omit it?

Reply:
done